tests/cli/test_cmds_spark_run.py

```diff
@@ -823,6 +835,130 @@ def test_configure_and_run_docker_driver_resource_limits(
+                    "spark.kubernetes.executor.volumes.hostPath.0.mount.path": "/k8s/volume0",
+                    "spark.kubernetes.executor.volumes.hostPath.0.options.path": "/k8s/volume0",
+                    "spark.kubernetes.executor.volumes.hostPath.1.mount.readOnly": "false",
+                    "spark.kubernetes.executor.volumes.hostPath.1.mount.path": "/k8s/volume1",
+                    "spark.kubernetes.executor.volumes.hostPath.1.options.path": "/k8s/volume1",
+                },
+                ["/k8s/volume0:/k8s/volume0:ro", "/k8s/volume1:/k8s/volume1:rw"],
+            ),
+        ],
+    )
+    def test_configure_and_run_docker_driver_resource_limits_memory_cap(
+        self,
+        mock_get_history_url,
+        mock_et_signalfx_url,
+        mock_get_docker_cmd,
+        mock_create_spark_config_str,
+        mock_get_webui_url,
+        mock_send_and_calculate_resources_cost,
+        mock_run_docker_container,
+        mock_get_username,
+        cluster_manager,
+        spark_args_volumes,
+        expected_volumes,
+    ):
+        mock_get_username.return_value = "fake_user"
+        spark_conf = {
+            "spark.app.name": "fake_app",
+            "spark.ui.port": "1234",
+            # resourceAdjustmentFactor * 100g > driverNodeMemory * (1 - overheadFactor)
+            # => 2 * 100g > 128g * (1 - 0.9)
+            "spark.driver.memory": "100g",
+            "spark.driver.cores": "2",
+            **spark_args_volumes,
+        }
+        mock_run_docker_container.return_value = 0
+
+        args = mock.MagicMock()
+        args.aws_region = "fake_region"
+        args.cluster = "fake_cluster"
+        args.cmd = "pyspark"
+        args.work_dir = "/fake_dir:/spark_driver"
+        args.dry_run = True
+        args.mrjob = False
+        args.nvidia = False
+        args.enable_compact_bin_packing = False
+        args.cluster_manager = cluster_manager
+        args.docker_cpu_limit = False
+        args.docker_memory_limit = False
+        args.docker_disk_limit = False
+        with mock.patch.object(
+            self.instance_config, "get_env_dictionary", return_value={"env1": "val1"}
+        ):
+            retcode = configure_and_run_docker_container(
+                args=args,
+                docker_img="fake-registry/fake-service",
+                instance_config=self.instance_config,
+                system_paasta_config=self.system_paasta_config,
+                aws_creds=("id", "secret", "token"),
+                spark_conf=spark_conf,
+                cluster_manager=cluster_manager,
+                pod_template_path="unique-run",
+            )
+
+        assert retcode == 0
+        expected_capped_memory = DEFAULT_SPARK_DRIVER_NODE_MEMORY_MB - max(
+            MIN_SPARK_DRIVER_MEMORY_OVERHEAD_MB,
+            int(
+                DEFAULT_SPARK_DRIVER_OVERHEAD_FACTOR
+                * DEFAULT_SPARK_DRIVER_NODE_MEMORY_MB
+            ),
+        )
```

Comment:
should this be a static value (or a value coming from the test params?)? this looks like the code in `_calculate_spark_driver_memory_overhead_mb` - so it seems a little off to re-write code that we're ostensibly testing in a test